pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -621,45 +624,33 @@ pub mod pallet {
 		type Message = GatewayMessage<T::Message, T::RouterId>;
 
 		fn process(msg: Self::Message) -> (DispatchResult, Weight) {
-			match msg {
-				GatewayMessage::Inbound {
-					domain_address,
-					message,
-					router_id,
-				} => {
-					let mut counter = 0;
-
-					let res = Self::process_inbound_message(
+			// The #[transactional] macro only works for functions that return a
+			// `DispatchResult` therefore, we need to manually add this here.
```

Comment:
Q: Any reason for going with the overhead of `with_transaction` and branching over the result instead of just using `#[transactional]` and returning `Ok(())`?

Reply:
Not sure what you mean here - do you mean changing the return of `process` to `DispatchResult`?

Reply:
We can't use the `#[transactional]` macro in this function, that is not supported. We can either change the signature as I suggested in my previous message, or go with the current approach.